include/uk/essentials.h

```diff
@@ -218,17 +218,40 @@ extern "C" {
 #define IN_RANGE(val, base, len)			\
 	(((val) >= (base)) && ((val) < (base) + (len)))
 
+/**
+ * Tests if range 0 is equal to range 1
+ */
+#define RANGE_ISEQUAL(base0, len0, base1, len1)		\
+	(((base0) == (base1))				\
+	 && (len0) == (len1))
+
+/**
+ * Tests if range 0 contains range 1
+ * This is the case when both of the following conditions are true:
+ *  - The start of range 1 is within range 0
+ *  - The end of range 1 is within range 0
+ *  NOTE: The expressions take into account that `base + len` points to
+ *        the first value that is outside of a range.
+ */
+#define RANGE_CONTAIN(base0, len0, base1, len1)				\
+	(IN_RANGE((base1), (base0), (len0))				\
+	 && (((base1) + (len1)) > (base0))				\
+	 && (((base1) + (len1)) <= ((base0) + (len0))))
```

Comment:
Wouldn't it be more expressive if these were replaced with the construct below?
```
IN_RANGE((base1) + (len1) - 1, (base0), (len0)))
```

Reply:
I was in fact also thinking about this but I wanted to avoid to imply any data type for the arguments. As soon as I subtract or add an int 1, this would not work anymore for floating point, for instance.